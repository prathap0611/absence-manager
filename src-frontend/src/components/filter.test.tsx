import { render, fireEvent } from '@testing-library/react';
import TableFilter from './filter';

describe('filter tests', () => {
    const setTypeFilter = jest.fn();
    const setDateFilter = jest.fn();
    const setCurrentPage = jest.fn();

    it('should render the page correctly', () => {
        const { asFragment } = render(
            <TableFilter
                typeFilter="vacation"
                setTypeFilter={setTypeFilter}
                dateFilter="03/11/2020"
                setDateFilter={setDateFilter}
                setCurrentPage={setCurrentPage}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('should set filter type and reset current page', () => {
        const { getByTestId } = render(
            <TableFilter
                typeFilter=""
                setTypeFilter={setTypeFilter}
                dateFilter=""
                setDateFilter={setDateFilter}
                setCurrentPage={setCurrentPage}
            />
        );

        const typeFilter = getByTestId('type-filter');
        fireEvent.change(typeFilter, { target: { value: 'vacation' } });

        expect(setTypeFilter).toBeCalledWith('vacation');
        expect(setCurrentPage).toBeCalledWith(0);
    });

    it('should date filter and reset current page', () => {
        const { getByTestId } = render(
            <TableFilter
                typeFilter=""
                setTypeFilter={setTypeFilter}
                dateFilter=""
                setDateFilter={setDateFilter}
                setCurrentPage={setCurrentPage}
            />
        );

        const dateFilter = getByTestId('date-filter');
        fireEvent.change(dateFilter, { target: { value: '2020-05-12' } });

        expect(setDateFilter).toBeCalledWith('2020-05-12');
        expect(setCurrentPage).toBeCalledWith(0);
    });
});
