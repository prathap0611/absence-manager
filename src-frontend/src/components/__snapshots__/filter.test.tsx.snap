// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter tests should render the page correctly 1`] = `
<DocumentFragment>
  <section
    class="filter-container"
  >
    <label
      for="type-filter"
    >
      Choose filter type: 
      <select
        class="type-filter"
        data-testid="type-filter"
        id="type-filter"
      >
        <option
          value=""
        >
          Filter by type
        </option>
        <option
          value="vacation"
        >
          Vacation
        </option>
        <option
          value="sickness"
        >
          Sickness
        </option>
      </select>
    </label>
    <label
      for="date-filter"
    >
      Choose filter date: 
      <input
        class="date-filter"
        data-testid="date-filter"
        id="date-filter"
        type="date"
        value="03/11/2020"
      />
    </label>
  </section>
</DocumentFragment>
`;
